### PR TITLE
Move Filtering and Pagination Logic into EntityList Component: closes#2424 

### PR DESCRIPTION
**Pull request by tn3rb**
- moves ALL filtering and pagination control into EntityList
- moves Loading, Error, and Empty State display into EntityList
- simplifies DatesList and TicketsList (cuz of above changes)
- extracts reducers and types from dates and tickets filter state hooks
- fixes filters (all work now)
- fixes pagination and simplifies logic

### CODE REVIEW

File: assets/src/application/ui/layout/entityList/CardList.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	return (
+		<div className='ee-entity-list__card-view'>
+			{entities.map((entity) => (
+				<CardView id={entity.id} key={entity.id} />
+			))}
+		</div>
+	);
+};
+
+export default CardList;
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/CardList.tsx]: Nice

File: assets/src/application/ui/layout/entityList/pagination/useEntityPagination.ts [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-	};
+	const onShowSizeChange: onShowSizeChangeFn = useCallback((newPageNumber, newPerPage) => {
+		// the pagination component will recalculate the page number
+		// if it goes out of range after changing the perPage value,
+		// so save that else we'll get no results returned
+		if (newPageNumber && newPageNumber !== pageNumber) {
+			setPageNumber(newPageNumber);
+		}
+		setPerPage(newPerPage);
+	}, []);
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/pagination/useEntityPagination.ts]: 👍

File: assets/src/domain/eventEditor/ui/datetimes/datesList/filterBar/controls/DisplayDatesControl.tsx [hunk 3]
```diff
@@ -26,15 +22,15 @@ const DisplayDatesControl: React.FC<DisplayDatesControlProps> = ({ displayDates,
...
 						label: __('start dates only'),
 					},
 					{
-						value: 'end',
+						value: DisplayDates.end,
 						label: __('end dates only'),
 					},
 					{
-						value: 'both',
+						value: DisplayDates.both,
```
Comment (manzoorwanijk) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/filterBar/controls/DisplayDatesControl.tsx]: Much better with `enum`

File: assets/src/application/ui/layout/entityList/EntityList.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	return (
+		<div className={'ee-entity-list'}>
+			<Title className='ee-entity-list__header' level={3}>
+				{headerText}
+			</Title>
+			<EntityListFilterBar entityFilters={entityFilters} filterState={filterState} listId={listId} />
+			{entityList}
+			<EntityPagination {...paginationProps} />
+			<Divider dashed />
+			{footer}
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/EntityList.tsx]: It means if there is an error loading  entities, the Add New button won't be displayed.

File: assets/src/domain/eventEditor/ui/datetimes/datesList/index.tsx [hunk 5]
```diff
@@ -1,33 +1,33 @@
...
+	const entityFilters = <DatesListEntityFilters {...entityFiltersProps} />;
+
+	return (
+		<EntityList
+			entities={filteredEntities}
+			entityFilters={entityFilters}
+			entityType={TypeName.datetimes}
+			CardView={DateCard}
+			TableView={TableView}
+			footer={<AddNewDateButton />}
```
Comment (manzoorwanijk) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/index.tsx]: IMHO, Add New button should not be a part of the `EntityList`

File: assets/src/application/ui/layout/entityList/EntityList.tsx (hunk 4 above)
Comment (tn3rb) [reply assets/src/application/ui/layout/entityList/EntityList.tsx]: that's how it was previously... it **does** appear when there are no entities returned however

File: assets/src/domain/eventEditor/ui/datetimes/datesList/index.tsx (hunk 5 above)
Comment (tn3rb) [reply assets/src/domain/eventEditor/ui/datetimes/datesList/index.tsx]: hmmm... maybe... that will complicate the layout a bit... I'll think about it
don't want to make that change as part of this PR though even if I do decide to go that route

**Pull request merged**